framework/include/utils/InputParameters.h

```diff
@@ -657,6 +657,41 @@ class InputParameters : public Parameters
    */
   std::set<std::string> reservedValues(const std::string & name) const;
 
+  /// Get/set a string representing the location (i.e. filename,linenum) in the input text for the
+  /// block containing parameters for this object.
+  std::string & blockLocation() { return _block_location; };
+
+  /// Get/set a string representing the full HIT parameter path from the input file (e.g.
+  /// "Mesh/foo") for the block containing parameters for this object.
+  std::string & blockFullpath() { return _block_fullpath; }
+  const std::string & blockFullpath() const { return _block_fullpath; }
+
+  /// Get/set a string representing the location in the input text the parameter originated from
+  /// (i.e. filename,linenum) for the given param.
+  const std::string & inputLocation(const std::string & param) const
+  {
+    return _params.at(param)._input_location;
```

Comment:
Per our discussion yesterday. Let's remove these unchecked `at()` methods from the code. The problem is that if they are called and we don't catch the exception a rank may mysteriously die leaving the user with no error message and instead they receive a cryptic error about MPI running into some problem. It's super nasty to track down. 

@friedmud suggested deriving from standard map and doing simple error checking on the `at()` method but that would be pretty invasive in that we'd have to force people to stop using std::map everywhere. It's not a terrible solution but it's just one more things that users have to look at and wonder why or alternatively run into error messages with CIVET if we enforce that check. I'll think about this some more.

In the meantime, there's nothing stopping you from creating a simple macro if you'd like, or just avoid using that method. I'd entertain either.